### PR TITLE
Create generic getunclaimedgas and rename wallet ones

### LINKED ISSUE #241: GetUnclaimedGas Generic
Perhaps we need extra calls on RPCServer  plugin such as generic `GetUnclaimedGas` call apart of just the the one linked to the wallet itself: https://github.com/neo-project/neo-modules/blob/d62b18bcea99186614c3043603fcaa8e6214ba77/src/RpcServer/RpcServer.Wallet.cs#L71-L81

The one linked to the account should be something like `GetWalletUnclaimedGas`

### PR DESCRIPTION
**Pull request by vncoelho**
closes #241

### CODE REVIEW

File: src/RpcServer/RpcServer.SmartContract.cs [hunk 1]
```diff
@@ -92,5 +96,25 @@ private JObject InvokeScript(JArray _params)
...
+            JObject json = new JObject();
+            UInt160 script_hash;
+            try
+            {
+                script_hash = address.ToScriptHash();
+            }
+            catch
+            {
+                script_hash = null;
+            }
```
Comment (vncoelho) [inline src/RpcServer/RpcServer.SmartContract.cs:113]: maybe need to check if it is invalid and return.

Comment (shargon) [reply src/RpcServer/RpcServer.SmartContract.cs:113]: I think that we should throw an exception

Comment (vncoelho) [reply src/RpcServer/RpcServer.SmartContract.cs:113]: Throw the exception and what.in the json return,@shargon?

Comment (vncoelho) [reply src/RpcServer/RpcServer.SmartContract.cs:113]: done

**Comment by nicolegys** [comment]
Great! 
`getunclaimedgas` `getwalletbalance` `getwalletunclaimedgas` all work well.
But should we put `getunclaimedgas`  in SmartContract.cs?

**Comment by vncoelho** [comment]
HI @nicolegys,

I am not sure. I thought about SmartContract because GAS is now a `NativeContract`.

File: src/RpcServer/RpcServer.SmartContract.cs [hunk 2]
```diff
@@ -92,5 +96,27 @@ private JObject InvokeScript(JArray _params)
...
+            if (script_hash == null)
+                throw new RpcException(-100, "Invalid address");
+            SnapshotView snapshot = Blockchain.Singleton.GetSnapshot();
+            json["unclaimed"] = NativeContract.NEO.UnclaimedGas(snapshot, script_hash, snapshot.Height + 1).ToString();
```
Comment (shargon) [inline src/RpcServer/RpcServer.SmartContract.cs:117]: why +1?

Comment (vncoelho) [reply src/RpcServer/RpcServer.SmartContract.cs:117]: I did not think about it, I just copied from here https://github.com/neo-project/neo-modules/blob/82147d5592ada1ad7fd0d2888762963800f8c0d7/src/RpcServer/RpcServer.Wallet.cs#L78

Comment (shargon) [reply src/RpcServer/RpcServer.SmartContract.cs:117]: Maybe because your "claim TX" always will enter in the next block, hahahah I don't know :P

Comment (vncoelho) [reply src/RpcServer/RpcServer.SmartContract.cs:117]: Do you think we need to change it now?

Comment (shargon) [reply src/RpcServer/RpcServer.SmartContract.cs:117]: it's ok for me

**Comment by vncoelho** [comment]
Ready to merge?

**Comment by joeqian10** [comment]
> HI @nicolegys,
> 
> I am not sure. I thought about SmartContract because GAS is now a `NativeContract` instead of UTXO based Asset.

@vncoelho Hi Vitor, maybe we could move `getunclaimedgas` to RpcServer.BlockChain.cs, since it's getting some info from the blockchain. Let me know what you guys think. @nicolegys @erikzhang @shargon @superboyiii

**Pull request merged**